cli/src/migrations.rs

```diff
@@ -0,0 +1,179 @@
+use color_eyre::owo_colors::OwoColorize;
+use eyre::Context;
+use sqlx::{Pool, Postgres, Row};
+use std::path::Path;
+
+pub struct Migrations<'a> {
+    path: &'a Path,
+}
+
+/// Methods for managing database migrations
+impl<'a> Migrations<'a> {
+    pub async fn new(path: &'a Path) -> eyre::Result<Self> {
+        tokio::fs::create_dir_all(path)
+            .await
+            .wrap_err("Failed to create migrations dir")?;
+
+        Ok(Self { path })
+    }
+
+    /// Applies database migrations based on the stored migration files and the current state
+    /// of the database.
+    ///
+    /// This function retrieves the latest applied migration ID, and determines which migrations
+    /// (if any) need to be applied. It then applies the pending migrations sequentially
+    /// and updates the `schema_migrations` table to record each migration.
+    pub async fn apply(&self, connection_string: String) -> eyre::Result<()> {
+        println!("{}", console::style("Applying migrations...").dimmed());
+
+        let all_migrations = self.migrations().await?;
+
+        let connection = sqlx::PgPool::connect(&connection_string)
+            .await
+            .wrap_err("Failed to connect to database")?;
+
+        self.ensure_migrations_table(&connection)
```

Comment:
@flyin I wouldn't mix application of migrations and init of the table, these seem to me to be different processes. And normally we want to init migrations meta even before we start applying migrations. Can we move the init to `init()` method, and make sure it's called outside?

Reply:
Done. Moved it into Sqldb::provision() routine